src/html/html-translation-extractor.ts

```diff
@@ -0,0 +1,24 @@
+import { Attribute, Text } from "./element-context";
+
+export interface AngularElement {
+  tagName: string;
+  attributes: Attribute[];
+  texts: Text[];
+  startPosition: number;
+}
+
+export interface HtmlTranslationExtractionContext {
+  emitError(message: string, position: number): void;
+  emitSuppressableError(message: string, position: number): void;
+  registerTranslation(translation: {
```

Comment:
The type of the `translation` parameter is used (and declared inline) several times, introducing a type alias would remove some duplication here (even if only in typings and not affecting generated JS):

```
export type Translation = {
    translationId: string;
    defaultText?: string;
    position: number;
}
```

Edit: I see this conflicts with the existing `Translation` class, but still defining a type (with a slightly different name, e.g. `TranslationUnit`) could be beneficial (and `Translation` class could implement it, too, if `id` is renamed to `translationId` in there also).